src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1178,11 +1197,6 @@ internal ValueTask<IBackendMessage> ReadMessage(bool async, DataRowLoadingMode d
 
                     throw;
                 }
-                catch (NpgsqlException e) when (e.InnerException is TimeoutException && _userCancellationRequested)
```

Comment:
I see you pushed down this check into NpgsqlReadBuffer... I think that's good.

Reply:
Yep, and it actually looks cleaner 😄